Studio sample （update）

851996006 opened the pull request.

luzhang06 committed:
Add files via upload

luzhang06 committed:
Update sample_analyze_layout.py

851996006 committed:
Update

1. Add a default URL
2. Change "Analyze local documents" to "Analyze documents at URL".
3. Add environment variable setting instructions for Linux and MacOS users, and more specific instructions on where exactly they can find this information.

851996006 committed:
Update

1. Add a default URL
2. Change "Analyze local documents" to "Analyze documents at URL".
3. Add environment variable setting instructions for Linux and MacOS users, and more specific instructions on where exactly they can find this information.

luzhang06 commented on sample_analyze_layout.py:
```diff
@@ -0,0 +1,189 @@
+DESCRIPTION:
+    This sample demonstrates how to extract text, tables, selection marks, and layout information from a document
+    given through a file.
+    
```
text, tables, figures and selection marks and document structure (e.g., sections) information from a document.

luzhang06 commented on sample_analyze_layout.py:
```diff
@@ -0,0 +1,189 @@
+    
+    Note that selection marks returned from begin_analyze_document(model_id="prebuilt-layout") do not return the text
+    associated with the checkbox. For the API to return this information, build a custom model to analyze the
+    checkbox and its text. See sample_build_model.py for more information.
```
For this part, recommend moving to selection mark extraction comments part. Not necessary to show up at the beginning of the sample.

luzhang06 commented on sample_analyze_layout.py:
```diff
@@ -0,0 +1,189 @@
+
+PREREQUISITES:
+    Necessary prerequisites are listed below.
+    To find more details, please click the "How-to guides" link (https://aka.ms/AApjann) to visit the Documentation.
```
the short link seems not pointing to the section, can you please double check?

luzhang06 commented on sample_analyze_layout.py:
```diff
@@ -0,0 +1,189 @@
+    To find more details, please click the "How-to guides" link (https://aka.ms/AApjann) to visit the Documentation.
+
+    -------Python and IDE------
+    1) Python3.7 or latter (https://www.python.org/) . Your Python installation should include pip (https://pip.pypa.io/en/stable/).
```
remove additional space

luzhang06 commented on sample_analyze_layout.py:
```diff
@@ -0,0 +1,189 @@
+    # Analyze a local document：
+    # Delete or comment out the part of "Analyze a document at a URL" above.
+    # Uncomment the following codes. Enter the actual file path in "path_to_sample_documents" variable.
+    # path_to_sample_documents = os.path.abspath(
```
path_to_sample_document - this should be only one document, we don't support multiple documents as input in an API call

luzhang06 commented on sample_analyze_layout.py:
```diff
@@ -0,0 +1,189 @@
+                if cell.bounding_regions:
+                    for region in cell.bounding_regions:
+                        print(f"...content on page {region.page_number} is within bounding polygon '{region.polygon}'")
+
```
per our offline discussion, please add "extract detected figures" sample code here as well.

luzhang06 commented on sample_analyze_layout.py:
```diff
@@ -0,0 +1,189 @@
+        if "Invalid request".casefold() in error.message.casefold():
+            print(f"Uh-oh! Seems there was an invalid request: {error}")
+        # Raise the error again
+        raise
```
Put some comments for next step, we can provide 2 links:
- Learn more about Layout model: https://aka.ms/di-layout
- Find more sample code: https://aka.ms/doc-intelligence-samples

851996006 committed:
Update sample_analyze_layout.py

851996006 committed:
Update sample_analyze_layout.py

851996006 committed:
update

1. Change names of links to be meaningful
2. Add a new section of code to analyze figures
3. Modify the code of the “poller” when analyzing the URL
3. New annotation: concepts of span and bounding polygon
4. Optimize text expression

851996006 committed:
Update sample_analyze_layout.py

1. 【Note about selection marks】 Move to the corresponding position
2. The short links have been reviewed
3.【 path-to-sample-documents】, remove all "s"
4. Change to: pip install azure-ai-documentintelligence==1.0.0b2
5. The comments are changed to a form that facilitates the "ctrl+/" operation of the entire paragraph.
6. Add concept links to the 【Analyze figures】 section
7. Add “Next steps”
8. Delete extra spaces

851996006 committed:
Update sample_analyze_layout.py

851996006 committed:
Update sample_analyze_layout.py

1、Change to：pip install azure-ai-documentintelligence --pre
2、Add link（https://aka.ms/train-your-custom-model） to the mark selection section

851996006 committed:
Update sample_analyze_layout.py

851996006 committed:
Update sample_analyze_layout.py

New comment: Analyze a document in Blob Storage

851996006 committed:
v4.0

851996006 committed:
v4.0

851996006 committed:
V4.0

851996006 committed:
v4.0

luzhang06 commented on sample_analyze_read.py:
```diff
@@ -0,0 +1,192 @@
+    #     poller = document_intelligence_client.begin_analyze_document(
+    #         "prebuilt-read",
+    #         analyze_request=f,
+    #         features=[DocumentAnalysisFeature.STYLE_FONT],
```
There's addon feature turned on here, optional, you need to either delete it or make some comments on it for addon capability；）

851996006 replied:
Is my understanding correct?
![image](https://github.com/luzhang06/sample/assets/161273051/e887ccab-2a16-4616-8a0f-58d23dc1280e)

luzhang06 commented on sample_analyze_read.py:
```diff
@@ -0,0 +1,192 @@
+    print("----Languages detected in the document----")
+    if result.languages is not None:
+        for language in result.languages:
+            print(f"Language code: '{language.locale}' with confidence {language.confidence}")
```
Based on these lines, recommend turn on language addon in the previous comment.

luzhang06 commented on sample_analyze_read.py:
```diff
@@ -0,0 +1,192 @@
+            if style.is_handwritten:
+                print("Found the following handwritten content: ")
+                print(",".join([result.content[span.offset : span.offset + span.length] for span in style.spans]))
+            if style.font_style:
```
Can you please help verify if free tier resource can use paid addon?

851996006 replied:
I asked customer service, and the feedback was: some require upgrading and paying to use it, and the free ones also have limits.
![image](https://github.com/luzhang06/sample/assets/161273051/065b5af8-5f17-4d85-a274-a8a12a522b14)
![image](https://github.com/luzhang06/sample/assets/161273051/344dc449-a0c8-4143-9ff2-01e375deb2db)

luzhang06 commented on sample_analyze_read.py:
```diff
@@ -0,0 +1,192 @@
+                for word in words:
+                    print(f"......Word '{word.content}' has a confidence of {word.confidence}")
+
+        # Analyze selection marks.
```
Read model should not have selection marks. It's only available in Layout and some prebuilt model.

851996006 replied:
Already deleted

luzhang06 commented on sample_analyze_read.py:
```diff
@@ -0,0 +1,192 @@
+    # Analyze paragraphs.
+    if result.paragraphs:
+        print(f"----Detected #{len(result.paragraphs)} paragraphs in the document----")
+        for paragraph in result.paragraphs:
```
can you please double check if paragraph.role applies to read model output? My understanding is this only applies to layout model.

851996006 replied:
https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/concept-read?view=doc-intel-4.0.0#paragraphs
The Read concept page shows the extraction of the paragraph, and the output of the code is the same as the "output" display in the document.

luzhang06 commented on sample_analyze_receipts.py:
```diff
@@ -0,0 +1,184 @@
+    receipts: AnalyzeResult = poller.result()
+
+    # [START analyze_receipts]
+    # To find more schemas, please refer to: https://aka.ms/azsdk/documentintelligence/receiptfieldschema
```
this link seems not work any more, should be something like this: https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/concept-receipt?view=doc-intel-4.0.0#field-extraction

851996006 replied:
Replaced with field-related links and placed field schema links in the description section.

luzhang06 commented on sample_analyze_tax_us_w2.py:
```diff
@@ -0,0 +1,324 @@
+
+    See fields found on a US W-2 tax form here:
+    https://aka.ms/azsdk/documentintelligence/taxusw2fieldschema
+
```
Same, the short links seem not work. Please double check. Also it's a good idea to put the field schema link in the description section, please apply it to other prebuilt models as well.

851996006 replied:
Already applied it to other prebuilt models

851996006 committed:
v4.0

851996006 committed:
update

851996006 committed:
V4.0

851996006 committed:
update

851996006 committed:
Update sample_analyze_read.py
